src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBuffer.cs

```diff
@@ -103,18 +104,18 @@ private void AppendValue(ViewBufferValue value)
         private ViewBufferPage GetCurrentPage()
         {
             ViewBufferPage page;
-            if (Pages.Count == 0)
+            if (_pages.Count == 0)
```

Comment:
Consider dropping the struct and just doing what you need in here. This code already has special cases in it for `Count == 0`, so you'd just be adding a few more cases in a few places. And then you don't need to implement/test another class with `IReadOnlyList<>` and an `IEnumerator`